interpreters/src/show.rs

```diff
@@ -0,0 +1,193 @@
+    ctx: Context,
+    plan: ShowPlan,
+    catalog_manager: C,
+}
+
+impl<C: Manager + 'static> ShowInterpreter<C> {
+    pub fn create(ctx: Context, plan: ShowPlan, catalog_manager: C) -> InterpreterPtr {
+        Box::new(Self {
+            ctx,
+            plan,
+            catalog_manager,
+        })
+    }
+}
+
+impl<C: Manager> ShowInterpreter<C> {
+    fn show_create(plan: ShowCreatePlan) -> Result<Output> {
+        let show_create = ShowCreateInterpreter::create(plan);
+        show_create.execute_show_create()
+    }
+
+    fn show_tables(ctx: Context, catalog_manager: C) -> Result<Output> {
+        let schema = get_default_schema(&ctx, &catalog_manager).context(FetchSchema)?;
+
+        let tables_names = schema
+            .all_tables()
+            .context(FetchTables)?
+            .iter()
+            .map(|t| t.name().to_string())
+            .collect::<Vec<_>>();
+
+        let schema = DataSchema::new(vec![Field::new(
+            SHOW_TABLES_COLUMN_SCHEMA,
+            DataType::Utf8,
+            false,
+        )]);
+        let record_batch = RecordBatch::try_new(
+            Arc::new(schema),
+            vec![Arc::new(StringArray::from(tables_names))],
+        )
+        .context(CreateRecordBatch)?;
+
+        let record_batch = record_batch.try_into().context(ToCommonRecordType)?;
+
+        Ok(Output::Records(vec![record_batch]))
+    }
+
+    fn show_databases(ctx: Context, catalog_manager: C) -> Result<Output> {
+        let catalog = get_default_catalog(&ctx, &catalog_manager).context(FetchCatalog)?;
+        let schema_names = catalog
+            .all_schemas()
+            .context(FetchDatabases)?
+            .iter()
+            .map(|t| t.name().to_string())
+            .collect::<Vec<_>>();
+
+        let schema = DataSchema::new(vec![Field::new(
+            SHOW_DATABASES_COLUMN_SCHEMA,
+            DataType::Utf8,
+            false,
+        )]);
+        let record_batch = RecordBatch::try_new(
+            Arc::new(schema),
+            vec![Arc::new(StringArray::from(schema_names))],
+        )
+        .context(CreateRecordBatch)?;
+
+        let record_batch = record_batch.try_into().context(ToCommonRecordType)?;
+
+        Ok(Output::Records(vec![record_batch]))
+    }
+}
+
+#[async_trait]
+impl<C: Manager> Interpreter for ShowInterpreter<C> {
+    async fn execute(self: Box<Self>) -> InterpreterResult<Output> {
+        match self.plan {
+            ShowPlan::ShowCreatePlan(t) => Self::show_create(t).context(ShowCreateTable),
+            ShowPlan::ShowTables => {
+                Self::show_tables(self.ctx, self.catalog_manager).context(ShowTables)
+            }
+            ShowPlan::ShowDatabase => {
+                Self::show_databases(self.ctx, self.catalog_manager).context(ShowDatabases)
+            }
+        }
+    }
+}
+
+fn get_default_catalog<C: Manager>(
+    ctx: &Context,
+    catalog_manager: &C,
+) -> crate::create::Result<Arc<dyn Catalog + Send + Sync>> {
```

Comment:
Shall we define the error in this mod so that there is no need to convert this error?